lib/response.js

```diff
@@ -20,7 +20,27 @@ function Response (req, onEnd) {
   this.once('finish', () => {
     const res = generatePayload(this)
     res.raw.req = req
-    process.nextTick(() => onEnd(res))
+    if (typeof reject === 'function') {
```

Comment:
If you're doing `new Promise((resolve, reject) => {})` when would `reject` ever _not_ be a function?

Reply:
In the case you are using callback style :)
https://github.com/fastify/light-my-request/blob/aeb6c202b12262f65a692d3d80a458561e4aa7b6/index.js#L67-L71

Reply:
Oh, I missed that first `if` case. 👍